applications/serial_lte_modem/src/slm_at_tcpip.c

```diff
@@ -490,6 +478,35 @@ static int do_send(const uint8_t *data, int datalen)
 	}
 }
 
+static int do_send_datamode(const uint8_t *data, int datalen)
+{
+	uint32_t offset = 0;
+	int ret = 0;
+	int sock = client.sock;
+
+	/* For TCP/TLS Server, send to imcoming socket */
+	if (client.role == AT_SOCKET_ROLE_SERVER) {
+		if (client.sock_peer != INVALID_SOCKET) {
+			sock = client.sock_peer;
+		} else {
+			LOG_ERR("No remote connection");
+			return -EINVAL;
+		}
+	}
+
+	while (offset < datalen) {
+		ret = send(sock, data + offset, datalen - offset, 0);
+		if (ret < 0) {
+			LOG_ERR("send() failed: %d", -errno);
+			ret = -errno;
```

Comment:
`ret` seems to be unused anyway? Should the error code be assigned to `offset` instead?

Reply:
Used as return value in the end

Reply:
Either I miss something, or there is no change.

Reply:
Sorry, it's copied from do_send() and indeed not used in this function, will be removed